### ISSUE TITLE
Accented characters

### ISSUE THREAD

**Issue by gloignon** [vanatteveldt/rsyntax#5]
When parsing a bit of text with accents, I get the wrong tokens. For instance, with the code:

```
tokens <- udpipe_tokenindex("Mon ami Hervé n'est pas en Provence, il est intelligent.", udpipe_model = "french-gsd")

tokens %>%
  plot_tree(token, lemma, POS)
```

![image](https://user-images.githubusercontent.com/12416756/70752948-b1009680-1d01-11ea-9c59-b533280bde1b.png)

Interestingly, when I use the standard udpipe parser, I can then use other functions in rsyntax, e.g.

![image](https://user-images.githubusercontent.com/12416756/70753944-fcb43f80-1d03-11ea-9aee-c3c4752abbcb.png)

**Comment by kasperwelbers** [comment]
Sorry for the late reply, I just found out that apparently all my Github notifications stopped working.

I can't seem to replicate the bug, but perhaps this has something to do with locale settings. Do I understand correctly that you did receive the correct output when you used udpipe directly, but not when you used udpipe_tokenindex?

**Comment by gloignon** [comment]
Correct. The accents work fine when I use the UDPipe parser and then modify the resulting table a bit so that it works with rsyntax. When using udpipe_tokenindex, accents turn into what seem like chinese characters, e.g:
![image](https://user-images.githubusercontent.com/12416756/72939062-24061f80-3d3a-11ea-83ba-d34bafe9fb19.png)

**Comment by kasperwelbers** [comment]
I'm afraid that I cannot reproduce the problem. On my system the output of udpipe (directly) and udpipe_tokenindex is both in UTF-8. 

Looking at your first post, it seems the problem already occurs in the input given to udpipe, because the accent on "herve" is considered a separate token. udpipe requires UTF-8, and I can more or less replicate your output if I convert the input to Latin1. Could you check whether your input to udpipe_tokensindex has UTF8 encoding (using `Encoding("your text")`). If not, you can try converting with iconv("your text", from = "your encoding", to='UTF-8'). Also, if this is the problem, it is generally recommended to set your locale to UTF8.

What I don't understand, though, is why your input to  rsyntax::udpipe_tokenindex and udpipe::udpipe_annotate would be different. Could you provide the code for how you compared the two?

**Comment by jwijffels** [comment]
Works fine for me. @gloignon You are probably working on an Windows computer with default Latin1 encoding, similar as me. When using udpipe, you need to make sure, the encoding of your text is UTF-8.

```
> library(rsyntax)
> library(magrittr)
> x <- "Mon ami Hervé n'est pas en Provence, il est intelligent."
> Encoding(x)
[1] "latin1"
> x <- iconv(x, from = 'latin1', to = 'UTF-8')
> Encoding(x)
[1] "UTF-8"
> tokens <- udpipe_tokenindex(x, udpipe_model = "french-gsd")
> 
> tokens %>%
+   plot_tree(token, lemma, POS)
Document: 1
Sentence: 1
```
![Rplot](https://user-images.githubusercontent.com/1710810/72972223-56ac2880-3dcb-11ea-90e7-287a693eb5a7.png)

**Comment by kasperwelbers** [comment]
Thanks for confirming @jwijffels!

I see that udpipe now gives a warning for non UTF-8 input, so I should probably add that here as well. 

By the way (sorry for going off topic), have you worked with or know of any implementations of coreference resolution for udpipe? For a current project where this is important I now need to use spacy, but it would be nice if I could do it in udpipe, which plays much nicer with R.

On that note, I should also really get around to getting this package on cran.

**Comment by jwijffels** [comment]
Never had chat-type of data where I had to do co-reference resolution. Don't know of anything on top of udpipe which does that. 
Maybe you could use the crfsuite R package (https://CRAN.R-project.org/package=crfsuite) to brew up a model in the lines of what is indicated at https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4067305
If you know of another C/C++ toolkit which does coreference resolution, I don't mind putting an R wrapper around it.

**Comment by kasperwelbers** [comment]
Thanks for the links. I didn't know crfsuite, cool stuff. 

For rsyntax it would be nice because with clause level analysis you often deal with anaphora. In an article about Trump, where "he" says or does something, it would be nice to link this back to Trump.

I am familiar with the gist of co-reference resolution, but every night I light a candle that a linguist has already implemented it for udpipe. If I find a nice C/C++ tool I might take a stab at it and let you know, but the big problem here would probably be finding suitable training data.

**Comment by jwijffels** [comment]
Yes, feel your grief on training data :)
FYI. Maybe these NLP package might as well be of your interest (not linked to your coreference resolution question though)
https://cran.r-project.org/web/packages/crfsuite/index.html
https://cran.r-project.org/web/packages/BTM/index.html
https://cran.r-project.org/web/packages/ruimtehol/index.html
https://cran.r-project.org/web/packages/tokenizers.bpe/index.html
https://cran.r-project.org/web/packages/textrank/index.html
https://github.com/bnosac/sentencepiece

**Comment by kasperwelbers** [comment]
Cool, thanks! I appreciate that you're bringing good c++ stuff into R :)

I never heard of BTM, but there's a lot of social media analysis in my field. Ruimtehol also looks really interesting.

**Comment by gloignon** [comment]
I fixed the encoding problem by adding as_utf8() around the phrase. E.g.:

```
phrase <- "Si Luc était un chat, il aurait mangé le poisson."
mes.tokens <- udpipe_tokenindex(as_utf8(phrase), udpipe_model = "french-gsd")
mes.tokens$pos <- mes.tokens$POS
```

Result:
![image](https://user-images.githubusercontent.com/12416756/72992400-aede2d00-3dc1-11ea-91e2-5be29197e2e8.png)

Looking good :)

Edit: I just found out that we can now set a UTF8 locale in Windows10, might look into that. But for the code to be reproducible on my colleges' computers, I might keep that as_utf8 in the code.
